remote/remote_test.go

```diff
@@ -1496,3 +1530,11 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 		})
 	})
 }
+
+func verifyRetryLogic(t *testing.T, mockServer *h.MockServer, repoName string, expectedCount int) {
+	defer mockServer.Server().Close()
```

Comment:
This might make more sense in an `it.After`

Reply:
I agreed with all the other suggestions, but this one makes me think.  If I refactor the code for actually exercising the SUT (System Under Test) and moves the verification to the `it.After` it will look like this:
```
when("manifest API in registry returns status code 200", func() {
	it.Before(func() {
		mockServer, repoName = h.SetUpMockServer(t, "org/do-not-retry", http.StatusOK, 0)
	})

	it("does not retry after status code 200", func() {
		exerciseRetryLogic(t, mockServer, repoName)
	})

	it.After(func() {
		h.AssertEq(t, mockServer.ActualCount(), 1)
	})
})

func exerciseRetryLogic(t *testing.T, mockServer *h.MockServer, repoName string) {
	defer mockServer.Server().Close()
	_, err := remote.NewImage(repoName, authn.DefaultKeychain, remote.WithPreviousImage(repoName))
	h.AssertNil(t, err)
}
```

I will not need the Assertion method, but I can keep the exercise in a different method to avoid duplication. How does it look like? @natalieparellano

Reply:
Oh sorry, I meant just `defer mockServer.Server().Close()` in the `After`